overlord/snapstate/progress.go

```diff
@@ -48,10 +50,21 @@ func NewTaskProgressAdapterLocked(t *state.Task) progress.Meter {
 func (t *taskProgressAdapter) Start(label string, total float64) {
 	t.label = label
 	t.total = total
+	t.Set(0.0)
 }
 
 // Set sets the current progress
 func (t *taskProgressAdapter) Set(current float64) {
+	t.current = current
+
+	// check if we made at least "minProgress" before we lock the state
+	const minProgress = 0.2 / 100.0
+	if current != 0.0 && (t.current-t.lastReported)/t.total < minProgress {
```

Comment:
In a hypothetical situation when `lastReported` is smaller than `current` this change will cause Set to no longer update the state. I would suggest to use the absolute value, so that whenever there's a change of 0.2%, in whichever direction, the status update takes place.

```suggestion
	if current != 0.0 && math.Abs(t.current-t.lastReported)/t.total < minProgress {
```

NOTE: You will need to import `math` here.